### ISSUE TITLE
There should be a way to visualise weapon ranges

### ISSUE THREAD

**Issue by opusforlife2** [endless-sky/endless-sky#10618]
### Problem Description

The game gives you the ability to shuffle around your weapons to different installation points on the ship. Combined with the variation in ship dimensions and weapon ranges, this leads to an optimisation problem where you need to figure out a way to maximise your offence and defence, not just numerically, but _spatially_.

For example, if you have a long Korath World Ship and you install just a short-ranged anti-missile turret at the very back, it can leave the front pretty vulnerable.

Similarly, if you have both short and long-range weapons, you might wish to place them such that they all converge near the same point/radius for maximum attack effect, or spread them out for maximum defence coverage.

Right now, you only have trial-and-error (or a spatially genius mind) as the sole method to test and optimise this.

### Related Issue Links

N/A

### Desired Solution

There's a ship diagram in Ship Info that shows where the different weapons are installed. I propose adding a separate tab and moving/copying the ship diagram there.

Here, if you hover over/click on a particular weapon's name, it would show the maximum range that weapon can reach from its installation point. For turrets, this would be a circle with the range as its radius, and the hardpoint as its centre. For guns, this would be a line showing the path of the projectile/beam from the hardpoint to the endpoint of its range.

Hover to show temporarily, click to show permanently (until clicked again to hide).

(Side note: this may also protect you from the immense embarrassment of installing massively powerful guns on a Palavret, only to realise they fire sideways, like a Wild West sheriff showing off their pistols.)

### Alternative Approaches

Instead of hovering/clicking as the range-reveal mechanism, there could be a filter panel with checkboxes, like in the Outfitter screen. Here you could pick and choose which weapons to show the ranges for, including by category (offence vs defence, guns vs turrets, primary vs secondary weapons, etc.).

**Comment by esksp** [comment]
> I propose adding a separate tab and moving/copying the ship diagram there.

That's a good idea: A separate "Weapon management panel"*, which shows enough empty space around your ship to show a color highlighted range circle for the currently selected weapon (gray for the others), allowing you to move your weapons around and see how their ranges interact.

I admit for some long-range weapons it would be nearly impossible to show range, but then again most weapons have a astonishingly small range, for some it's almost just a (big) ship's length. And for those very long range weapons placement isn't very important, is it.

Given you don't need to move your weapons around very often, it doesn't matter it's in a submenu, what matters is the increased efficiency.

*- It would also allow to make the whole weapon management thing bigger, since it wouldn't have to _also_ display the rest of the ship's data.

**Comment by esksp** [comment]
Good idea, but IMHO unimportant since you wouldn't really care where on your ship you put those. It's only important for the tiny ranged ones (and your personal need for symmetry and tidiness...).
